### DISCUSSION TITLE
Measuring greater voltage signals

### DISCUSSION THREAD

**Discussion by Abhijeetbyte** [fhdm-dev/scoppy#71] [Scoppy Front end designs]
Greeting!

As you mentioned, it measures signals between 0V and 3.3V, the signal voltage must be within the allowed range of the ADC.
Hence to measure the high voltage signal peak to peak, a voltage divider circuit is proposed here.<br/>

It uses 900k, 90k, and 9k resistors and a rotary switch to change between voltage levels.
The purpose here is not to feed voltage signals above the limits of the ADC and to know the actual voltage, only the measured voltage needs to be multiplied by the corresponding labeled value (x1, x10, x100).<br/>

I haven't done my testing with this circuit yet, I will soon.
What you think about this!

Thankyou <br/>

![WhatsApp Image 2022-07-16 at 11 56 14 AM](https://user-images.githubusercontent.com/80936610/179343583-78f747d8-8849-46d4-b0d0-bdadba8930f0.jpeg)

**Comment by Jason-BCIT** [comment]
I'm using a similar technique to attenuate and offset the input signal. I've placed a pot between the two 1K resistors. The voltage coming off the junction of the two resistors as the voltage coming off that point actually needs to be a bit higher than 1.65V in order to get correct readings from your voltage divider... I find that somewhere around 1.73V works well.

It seems counterintuitive that you need more than a 1.65V offset, but your output to GP26 should be 1.65V when Vin is at 0V.... that means that there has to be a small voltage drop across the attenuation resistors. After all, if Vin is zero, and the output of the voltage divider is 1.65V, you're going to have a voltage drop across the 9K (and/or 90K) before the signal gets to GP26. You can play around with this using Falstad's amazing online circuit simulator... https://tinyurl.com/2xu4awe4   As you change the switch (and thus the attenuation ratio) you may find that your zero value shifts... you'll have to ground the Vin line and tweak the pot to get a true zero before beginning your readings.

Hope that helps,

Jason
